common/core/test/_shared_access_signature_test.js

```diff
@@ -175,6 +176,121 @@ describe('SharedAccessSignature', function () {
+      throwsWhenFalsy('expiry');
+    });
+
+    /*Tests_SRS_NODE_COMMON_SAS_06_001: [If `credentials`, `expiry`, `signingFunction`, or `callback` are falsy, `createWithSigningFunction` shall throw ReferenceException.] */
+    it('throws when \'signingFunction\' is falsy', function () {
+      throwsWhenFalsy('signingFunction');
+    });
+
+    /*Tests_SRS_NODE_COMMON_SAS_06_001: [If `credentials`, `expiry`, `signingFunction`, or `callback` are falsy, `createWithSigningFunction` shall throw ReferenceException.] */
+    it('throws when \'callback\' is falsy', function () {
+      throwsWhenFalsy('callback');
+    });
+
+    function throwsWhenFalsy(argName) {
+      var args = {
+        credentials: {a: 'b'},
+        expiry: 1,
+        signingFunction: () => {},
+        callback: () => {}
+      };
+
+      function _createWithSigningFunction() {
+        return SharedAccessSignature.createWithSigningFunction(args.credentials, args.expiry, args.signingFunction, args.callback);
+      }
+
+      function throws(element) {
+        args[argName] = element;
+        var messageMatch = new RegExp('^Argument \'' + argName + '\' is ' + element);
+        assert.throws(_createWithSigningFunction, ReferenceError, messageMatch);
+      }
+
+      [undefined, null, '', 0].forEach(throws);
+    }
+
+    /*Tests_SRS_NODE_COMMON_SAS_06_002: [The `createWithSigningFunction` shall create a `SharedAccessSignature` object with an `sr` property formed by url encoding `credentials.host` + `/devices/` + `credentials.deviceId`.] */
+    it('creates a valid sr value', function(callback) {
+      var fakeCredentials = {
+        host: 'fakeHostName',
+        deviceId: 'fakeDeviceId'
+      };
+      var fakeSignedValue = Buffer.from('abcd');
+      var fakeSigningFunction = sinon.stub().callsArgWith(1, null, fakeSignedValue);
+      SharedAccessSignature.createWithSigningFunction(fakeCredentials, 1, fakeSigningFunction, (err, newSas) => {
+        assert.strictEqual(newSas.sr, 'fakeHostName%2Fdevices%2FfakeDeviceId', 'Invalid scope create for sas token');
+        assert.isUndefined(newSas.skn, 'Skn is improperly included');
+        callback();
+      });
+    });
+
+    /*Tests_SRS_NODE_COMMON_SAS_06_003: [** The `createWithSigningFunction` shall create a `SharedAccessSignature` object with an `se` property containing the value of the parameter `expiry`.] */
+    it('creates a valid se value', function(callback) {
+      var fakeCredentials = {
+        host: 'fakeHostName',
+        deviceId: 'fakeDeviceId'
+      };
+      var fakeSignedValue = Buffer.from('abcd');
+      var fakeSigningFunction = sinon.stub().callsArgWith(1, null, fakeSignedValue);
+      SharedAccessSignature.createWithSigningFunction(fakeCredentials, 1, fakeSigningFunction, (err, newSas) => {
+        assert.strictEqual(newSas.se, 1, 'Invalid expiry create for sas token');
+        callback();
+      });
+    });
+
+    it('creates a valid skn value', function(callback) {
```

Comment:
not requirement for this one?

Reply:
No, I just forgot to copy it.

---
In reply to: [165244629](https://github.com/Azure/azure-iot-sdk-node/pull/217#discussion_r165244629) [](ancestors = 165244629)